client/src/app.js

```diff
@@ -0,0 +1,144 @@
+import io from "socket.io-client";
+
+const ENDPOINT = "http://www.toonin.ml:8100/";
+
+const servers = {
+    iceServers: [
+      {
+        urls: [
+          "stun:stun.l.google.com:19302",
+          "stun:stun2.l.google.com:19302",
+          "stun:stun3.l.google.com:19302",
+          "stun:stun4.l.google.com:19302"
+        ]
+      }
+    ]
+};
+
+var socket;
+var audioElem;
+
+export function init(audioElement) {
+    audioElem = audioElement;
+    setSocketListeners = setSocketListeners.bind(this);
+    createAnswer = createAnswer.bind(this);
+}
+
+export function toonin() {
+    return;
+}
+
+export function logMessage(msg) { console.log(msg); }
+
+export function checkstream() {
+    var obj = {
+        room: this.room,
+        isPlaying: this.isPlaying,
+        established: this.established,
+        rtcConn: this.rtcConn,
+        peerID: this.peerID
+    };
+    fetch(ENDPOINT + this.room)
+        .then(res => res.json())
+        .then(res => checkStreamResult(res, obj))
+        .then(() => {
+            this.room = obj.room,
+            this.isPlaying = obj.isPlaying,
+            this.established = obj.established,
+            this.rtcConn = obj.rtcConn,
+            this.peerID= obj.peerID
+        })
+        .catch(err => logMessage(err));
+}
+
+export function checkStreamResult(result, obj) {
```

Comment:
We could split this function into onRoomFound and onNoSuchRoom. Also modified the passed in obj isn't super ideal.

Reply:
Yeah, there's a lot to change, especially with the obj. I just hacked some stuff together to get it running. It needs some refactoring